File: VB/App_Code/NumberMonthConverter.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web.UI


Public NotInheritable Class NumberMonthConverter

    Private Sub New()
    End Sub

    Private Shared Months As New List(Of Pair)() From { _
        New Pair("January", 1), _
        New Pair("Feburary", 2), _
        New Pair("March", 3), _
        New Pair("April", 4), _
        New Pair("May", 5), _
        New Pair("June", 6), _
        New Pair("July", 7), _
        New Pair("August", 8), _
        New Pair("September", 9), _
        New Pair("October", 10), _
        New Pair("November", 11), _
        New Pair("December", 12) _
    }
    Public Shared Function GetTitleOfMonth(ByVal numberOfMonth As Integer) As String
        For i As Integer = 0 To Months.Count - 1
            If numberOfMonth = Convert.ToInt32(Months(i).Second) Then
                Return Months(i).First.ToString()
            End If
        Next i
        Return String.Empty
    End Function
End Class

File: CS/App_Code/NumberMonthConverter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web.UI;


public static class NumberMonthConverter {
    static private List<Pair> Months = new List<Pair>() {
        new Pair("January", 1), 
        new Pair("Feburary", 2),
        new Pair("March", 3),
        new Pair("April", 4),
        new Pair("May", 5),
        new Pair("June", 6),
        new Pair("July", 7),
        new Pair("August", 8),
        new Pair("September", 9),
        new Pair("October", 10),
        new Pair("November", 11),
        new Pair("December", 12) 
    };
    public static string GetTitleOfMonth(int numberOfMonth) {
        for (int i = 0; i < Months.Count; i++)
            if (numberOfMonth == Convert.ToInt32(Months[i].Second))
                return Months[i].First.ToString();
        return string.Empty;
    }
}